### PR TITLE
Add cache_file parameter for DynamicClient

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

-  Add logic for cache file name generation
-  Add cache_file when DynamicClient is created

### CODE REVIEW

File: plugins/module_utils/common.py [hunk 1]
```diff
@@ -181,8 +218,14 @@ def auth_set(*names):
...
 
+    def generate_cache_file(configuration):
+        return 'osrcp-{0}.json'.format(hashlib.sha1(get_default_cache_id(configuration)).hexdigest())
+
+    kubeclient = kubernetes.client.ApiClient(configuration)
+    cache_file = generate_cache_file(kubeclient)
+
     try:
-        client = DynamicClient(kubernetes.client.ApiClient(configuration))
+        client = DynamicClient(kubernetes.client.ApiClient(configuration), cache_file)
```
Comment (Akasurde) [inline plugins/module_utils/common.py:228]:
```suggestion
        client = DynamicClient(kubeclient, cache_file)
```

Could you please add test case and changelog for this change? Thanks

Comment (alinabuzachis) [reply plugins/module_utils/common.py]: Yes, sure. I also applied this change. Thanks.

**Comment by Akasurde** [comment]
@alinabuzachis I am getting this - 

```
    cache_id = "{0}-{1}".format(configuration.host, user)
AttributeError: 'ApiClient' object has no attribute 'host'
```

**Comment by gravesm** [comment]
Closing as this has been reopened in https://github.com/ansible-collections/kubernetes.core/pull/46.

**Pull request closed without merging**